crates/oxc_aria_query/src/lib.rs

```diff
@@ -0,0 +1,771 @@
+    "sectionhead", "select", "structure", "widget", "window",
+};
+
+pub static INTERACTIVE_ROLES: phf::Set<&'static str> = phf_set! {
+    "button", "checkbox", "columnheader", "combobox", "doc-backlink", "doc-biblioref",
+    "doc-glossref", "doc-noteref", "grid", "gridcell", "link", "listbox", "menu",
+    "menubar", "menuitem", "menuitemcheckbox", "menuitemradio", "option", "radio",
+    "radiogroup", "row", "rowheader", "scrollbar", "searchbox", "slider", "spinbutton",
+    "switch", "tab", "tablist", "textbox", "tree", "treegrid", "treeitem", "toolbar",
+};
+
+pub static NONINTERACTIVE_ROLES: phf::Set<&'static str> = phf_set! {
+    "alert", "alertdialog", "application", "article", "banner", "blockquote", "caption",
+    "cell", "code", "complementary", "contentinfo", "definition", "deletion", "dialog",
+    "directory", "doc-abstract", "doc-acknowledgments", "doc-afterword", "doc-appendix",
+    "doc-biblioentry", "doc-bibliography", "doc-chapter", "doc-colophon",
+    "doc-conclusion", "doc-cover", "doc-credit", "doc-credits", "doc-dedication",
+    "doc-endnote", "doc-endnotes", "doc-epigraph", "doc-epilogue", "doc-errata",
+    "doc-example", "doc-footnote", "doc-foreword", "doc-glossary", "doc-index",
+    "doc-introduction", "doc-notice", "doc-pagebreak", "doc-pagefooter",
+    "doc-pageheader", "doc-pagelist", "doc-part", "doc-preface", "doc-prologue",
+    "doc-pullquote", "doc-qna", "doc-subtitle", "doc-tip", "doc-toc", "document",
+    "emphasis", "feed", "figure", "form", "graphics-document", "graphics-object",
+    "graphics-symbol", "group", "heading", "img", "insertion", "list", "listitem", "log",
+    "main", "mark", "marquee", "math", "meter", "navigation", "none", "note",
+    "paragraph", "presentation", "region", "rowgroup", "search", "separator", "status",
+    "strong", "subscript", "superscript", "table", "tabpanel", "term", "time", "timer",
+    "tooltip", "progressbar",
+};
+
+pub static INTERACTIVE_ELEMENT_SCHEMA: &[ElementSchema] = &[
+    ElementSchema {
+        name: "a",
+        attributes: &[Attr { name: "href", value: None }],
+    },
+    ElementSchema {
+        name: "area",
+        attributes: &[Attr { name: "href", value: None }],
+    },
+    ElementSchema {
+        name: "button",
+        attributes: &[],
+    },
+    ElementSchema {
+        name: "datalist",
+        attributes: &[],
+    },
+    ElementSchema {
+        name: "input",
+        attributes: &[
+            Attr {
+                name: "type",
+                value: Some("button"),
+            },
+        ],
+    },
+    ElementSchema {
+        name: "input",
+        attributes: &[
+            Attr {
+                name: "type",
+                value: Some("image"),
+            },
+        ],
+    },
+    ElementSchema {
+        name: "input",
+        attributes: &[
+            Attr {
+                name: "type",
+                value: Some("reset"),
+            },
+        ],
+    },
```

Comment:
I know this is more or less a direct port of `aria-query`, but it seems like these values are stored suboptimally. If we're only interested in `input` element schemas, we have to iterate over the entire array to find them. It seems like it would be better to store separate arrays for each element name, and iterate over those arrays instead. That would also allow us to easily check if a given tag is in the schemas list too.

Something more like this:

```
  "input" => &[
    ElementSchema {
      attributes: &[/* ... */]
    },
    ElementSchema {
      attributes: &[/* ... */]
    },
  ],
```